application_sandbox/shader_16bit_storage/main.cpp

```diff
@@ -0,0 +1,376 @@
+    VK_FALSE};
+
+// This creates an application with 16MB of image memory, and defaults
+// for host, and device buffer sizes.
+class CubeSample : public sample_application::Sample<CubeFrameData> {
+ public:
+  CubeSample(const entry::EntryData* data,
+             const VkPhysicalDeviceFeatures& request_features)
+      : data_(data),
+        Sample<CubeFrameData>(
+            data->allocator(), data, 1, 512, 1, 1,
+            sample_application::SampleOptions().AddDeviceExtensionStructure(
+                &sixteen_bit_storage_feature),
+            {request_features},
+            {VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME},
+            {VK_KHR_16BIT_STORAGE_EXTENSION_NAME}),
+        cube_(data->allocator(), data->logger(), cube_data) {}
+  virtual void InitializeApplicationData(
+      vulkan::VkCommandBuffer* initialization_buffer,
+      size_t num_swapchain_images) override {
+    cube_.InitializeData(app(), initialization_buffer);
+
+    VkPhysicalDevice16BitStorageFeatures ext_features = {
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_16BIT_STORAGE_FEATURES};
+    VkPhysicalDeviceFeatures2 features = {
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2, &ext_features};
+    app()->instance()->vkGetPhysicalDeviceFeatures2KHR(
+        app()->device().physical_device(), &features);
+    app()->instance()->GetLogger()->LogInfo("VK_KHR_16bit_storage features:");
+    app()->instance()->GetLogger()->LogInfo(
+        "storageBuffer16BitAccess: ", ext_features.storageBuffer16BitAccess);
+    app()->instance()->GetLogger()->LogInfo(
+        "uniformAndStorageBuffer16BitAccess: ",
+        ext_features.uniformAndStorageBuffer16BitAccess);
+    app()->instance()->GetLogger()->LogInfo("storagePushConstant16: ",
+                                            ext_features.storagePushConstant16);
+    app()->instance()->GetLogger()->LogInfo("storageInputOutput16: ",
+                                            ext_features.storageInputOutput16);
+
+    cube_descriptor_set_layouts_[0] = {
+        0,                                  // binding
+        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,  // descriptorType
+        1,                                  // descriptorCount
+        VK_SHADER_STAGE_VERTEX_BIT,         // stageFlags
+        nullptr                             // pImmutableSamplers
+    };
+    cube_descriptor_set_layouts_[1] = {
+        1,                                  // binding
+        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,  // descriptorType
+        1,                                  // descriptorCount
+        VK_SHADER_STAGE_VERTEX_BIT,         // stageFlags
+        nullptr                             // pImmutableSamplers
+    };
+    cube_descriptor_set_layouts_[2] = {
+        2,                                  // binding
+        VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,  // descriptorType
+        1,                                  // descriptorCount
+        VK_SHADER_STAGE_VERTEX_BIT,         // stageFlags
```

Comment:
Isn't this one going VK_SHADER_STAGE_FRAGMENT_BIT?

Reply:
Thanks for the catch! I was using it in vertex shader at first and then forgot to change.